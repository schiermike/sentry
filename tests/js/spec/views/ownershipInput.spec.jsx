import React from 'react';
import {mount} from 'enzyme';

import {ThemeProvider} from 'emotion-theming';
import theme from 'app/utils/theme';
import OwnerInput from 'app/views/settings/project/projectOwnership/ownerInput';

jest.mock('jquery');
describe('Project Ownership Input', function() {
  let org;
  let project;
  let put;

  beforeEach(function() {
    org = TestStubs.Organization();
    project = TestStubs.Project();

    MockApiClient.addMockResponse({
      url: '/organizations/org-slug/members/',
      method: 'GET',
      body: [TestStubs.Members()],
    });
    put = MockApiClient.addMockResponse({
      url: `/projects/${org.slug}/${project.slug}/ownership/`,
      method: 'PUT',
      body: {raw: 'url:src @dummy@example.com'},
    });
  });

  it('renders', function() {
    let wrapper = mount(
      <ThemeProvider theme={theme}>
        <OwnerInput
          params={{orgId: org.slug, projectId: project.slug}}
          organization={org}
          initialText="url:src @dummy@example.com"
          project={project}
        />
      </ThemeProvider>,
      TestStubs.routerContext()
    );

    let submit = wrapper.find('SaveButton button');

    expect(put).not.toHaveBeenCalled();

    // if text is unchanged, submit button is disabled
    submit.simulate('click');
    expect(put).not.toHaveBeenCalled();

    wrapper
      .find(OwnerInput)
      .instance()
      .onChange({target: {value: 'new'}});

    submit.simulate('click');

    expect(put).toHaveBeenCalled();

    expect(wrapper.find(OwnerInput)).toMatchSnapshot();
  });
});
