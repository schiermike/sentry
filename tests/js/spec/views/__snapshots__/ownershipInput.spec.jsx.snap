// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Ownership Input renders 1`] = `
<OwnerInput
  initialText="url:src @dummy@example.com"
  organization={
    Object {
      "access": Array [
        "org:read",
        "org:write",
        "org:admin",
        "project:read",
        "project:write",
        "project:admin",
        "team:read",
        "team:write",
        "team:admin",
      ],
      "features": Array [],
      "id": "3",
      "name": "Organization Name",
      "onboardingTasks": Array [],
      "projects": Array [],
      "slug": "org-slug",
      "status": Object {
        "id": "active",
        "name": "active",
      },
      "teams": Array [],
    }
  }
  params={
    Object {
      "orgId": "org-slug",
      "projectId": "project-slug",
    }
  }
  project={
    Object {
      "hasAccess": true,
      "id": "2",
      "isBookmarked": false,
      "isMember": true,
      "name": "Project Name",
      "slug": "project-slug",
      "teams": Array [],
    }
  }
>
  <RuleBuilder
    onAddRule={[Function]}
    organization={
      Object {
        "access": Array [
          "org:read",
          "org:write",
          "org:admin",
          "project:read",
          "project:write",
          "project:admin",
          "team:read",
          "team:write",
          "team:admin",
        ],
        "features": Array [],
        "id": "3",
        "name": "Organization Name",
        "onboardingTasks": Array [],
        "projects": Array [],
        "slug": "org-slug",
        "status": Object {
          "id": "active",
          "name": "active",
        },
        "teams": Array [],
      }
    }
    project={
      Object {
        "hasAccess": true,
        "id": "2",
        "isBookmarked": false,
        "isMember": true,
        "name": "Project Name",
        "slug": "project-slug",
        "teams": Array [],
      }
    }
  >
    <BuilderBar>
      <div
        className="css-1rsoebt-BuilderBar css-1i3lhlw5"
      >
        <BuilderSelect
          onChange={[Function]}
          showSearch={false}
          value="path"
        >
          <SelectInput
            className="css-1eslqkc-BuilderSelect css-1i3lhlw6"
            disabled={false}
            multiple={false}
            onChange={[Function]}
            placeholder="Select an option..."
            required={false}
            showSearch={false}
            value="path"
          >
            <select
              className="css-1eslqkc-BuilderSelect css-1i3lhlw6"
              disabled={false}
              multiple={false}
              onChange={[Function]}
              placeholder="Select an option..."
              required={false}
              value="path"
            >
              <option
                value="path"
              >
                Path
              </option>
              <option
                value="url"
              >
                URL
              </option>
            </select>
          </SelectInput>
        </BuilderSelect>
        <BuilderInput
          controlled={true}
          onChange={[Function]}
          placeholder="src/example/*"
          value=""
        >
          <input
            className="css-1t4syiu-Input-BuilderInput css-1i3lhlw7"
            onChange={[Function]}
            placeholder="src/example/*"
            value=""
          />
        </BuilderInput>
        <Divider
          src="icon-chevron-right"
        >
          <InlineSvg
            className="css-a4tn6s-Divider css-1i3lhlw8"
            src="icon-chevron-right"
          >
            <StyledSvg
              className="css-a4tn6s-Divider css-1i3lhlw8"
              height="1em"
              viewBox={Object {}}
              width="1em"
            >
              <svg
                className="css-1i3lhlw8 css-5e2rei-StyledSvg e2idor0"
                height="1em"
                viewBox={Object {}}
                width="1em"
              >
                <use
                  href="#test"
                  xlinkHref="#test"
                />
              </svg>
            </StyledSvg>
          </InlineSvg>
        </Divider>
        <Flex
          align="center"
          flex="1"
          mr={1}
        >
          <Base
            align="center"
            className="css-1q48dzo"
            flex="1"
            mr={1}
          >
            <div
              className="css-1q48dzo"
              is={null}
            >
              <SelectOwners
                onChange={[Function]}
                organization={
                  Object {
                    "access": Array [
                      "org:read",
                      "org:write",
                      "org:admin",
                      "project:read",
                      "project:write",
                      "project:admin",
                      "team:read",
                      "team:write",
                      "team:admin",
                    ],
                    "features": Array [],
                    "id": "3",
                    "name": "Organization Name",
                    "onboardingTasks": Array [],
                    "projects": Array [],
                    "slug": "org-slug",
                    "status": Object {
                      "id": "active",
                      "name": "active",
                    },
                    "teams": Array [],
                  }
                }
                project={
                  Object {
                    "hasAccess": true,
                    "id": "2",
                    "isBookmarked": false,
                    "isMember": true,
                    "name": "Project Name",
                    "slug": "project-slug",
                    "teams": Array [],
                  }
                }
                value={Array []}
              >
                <MultiSelectField
                  async={true}
                  cache={false}
                  defaultOptions={true}
                  disabled={false}
                  filterOptions={[Function]}
                  hideErrorMessage={false}
                  loadOptions={[Function]}
                  onChange={[Function]}
                  placeholder="Add Owners"
                  required={false}
                  style={
                    Object {
                      "width": 200,
                    }
                  }
                  value={Array []}
                  valueComponent={[Function]}
                >
                  <MultiSelect
                    arrowRenderer={[Function]}
                    async={true}
                    cache={false}
                    defaultOptions={true}
                    disabled={false}
                    filterOptions={[Function]}
                    hideErrorMessage={false}
                    id="id-undefined"
                    loadOptions={[Function]}
                    multi={true}
                    onChange={[Function]}
                    placeholder="Add Owners"
                    required={false}
                    style={
                      Object {
                        "width": 200,
                      }
                    }
                    value={Array []}
                    valueComponent={[Function]}
                  >
                    <Component
                      arrowRenderer={[Function]}
                      async={true}
                      cache={false}
                      className="css-1yznv6y-MultiSelect css-sbd3w60"
                      defaultOptions={true}
                      disabled={false}
                      filterOptions={[Function]}
                      hideErrorMessage={false}
                      id="id-undefined"
                      loadOptions={[Function]}
                      multi={true}
                      onChange={[Function]}
                      placeholder="Add Owners"
                      required={false}
                      style={
                        Object {
                          "width": 200,
                        }
                      }
                      value={Array []}
                      valueComponent={[Function]}
                    >
                      <Async
                        arrowRenderer={[Function]}
                        autoload={true}
                        cache={false}
                        className="css-1yznv6y-MultiSelect css-sbd3w60"
                        defaultOptions={true}
                        disabled={false}
                        filterOptions={[Function]}
                        hideErrorMessage={false}
                        id="id-undefined"
                        ignoreAccents={true}
                        ignoreCase={true}
                        loadOptions={[Function]}
                        loadingPlaceholder="Loading..."
                        multi={true}
                        onChange={[Function]}
                        options={Array []}
                        placeholder="Add Owners"
                        required={false}
                        searchPromptText="Type to search"
                        style={
                          Object {
                            "width": 200,
                          }
                        }
                        value={Array []}
                        valueComponent={[Function]}
                      >
                        <Select
                          arrowRenderer={[Function]}
                          autoload={true}
                          autosize={true}
                          backspaceRemoves={true}
                          backspaceToRemoveMessage="Press backspace to remove {label}"
                          cache={false}
                          className="css-1yznv6y-MultiSelect css-sbd3w60"
                          clearAllText="Clear all"
                          clearRenderer={[Function]}
                          clearValueText="Clear value"
                          clearable={true}
                          closeOnSelect={true}
                          defaultOptions={true}
                          deleteRemoves={true}
                          delimiter=","
                          disabled={false}
                          escapeClearsValue={true}
                          filterOptions={[Function]}
                          hideErrorMessage={false}
                          id="id-undefined"
                          ignoreAccents={true}
                          ignoreCase={true}
                          inputProps={Object {}}
                          isLoading={true}
                          joinValues={false}
                          labelKey="label"
                          loadOptions={[Function]}
                          loadingPlaceholder="Loading..."
                          matchPos="any"
                          matchProp="any"
                          menuBuffer={0}
                          menuRenderer={[Function]}
                          multi={true}
                          noResultsText="Loading..."
                          onBlurResetsInput={true}
                          onChange={[Function]}
                          onCloseResetsInput={true}
                          onInputChange={[Function]}
                          onSelectResetsInput={true}
                          openOnClick={true}
                          optionComponent={[Function]}
                          options={Array []}
                          pageSize={5}
                          placeholder="Loading..."
                          removeSelected={true}
                          required={false}
                          rtl={false}
                          scrollMenuIntoView={true}
                          searchPromptText="Type to search"
                          searchable={true}
                          simpleValue={false}
                          style={
                            Object {
                              "width": 200,
                            }
                          }
                          tabSelectsValue={true}
                          trimFilter={true}
                          value={Array []}
                          valueComponent={[Function]}
                          valueKey="value"
                        >
                          <div
                            className="Select css-1yznv6y-MultiSelect css-sbd3w60 is-clearable is-loading is-searchable Select--multi"
                          >
                            <div
                              className="Select-control"
                              onKeyDown={[Function]}
                              onMouseDown={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              style={
                                Object {
                                  "width": 200,
                                }
                              }
                            >
                              <span
                                className="Select-multi-value-wrapper"
                                id="react-select-2--value"
                              >
                                <div
                                  className="Select-placeholder"
                                >
                                  Loading...
                                </div>
                                <AutosizeInput
                                  aria-activedescendant="react-select-2--value"
                                  aria-expanded="false"
                                  aria-haspopup="false"
                                  aria-owns=""
                                  className="Select-input"
                                  id="id-undefined"
                                  injectStyles={true}
                                  minWidth="5"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  required={false}
                                  role="combobox"
                                  value=""
                                >
                                  <div
                                    className="Select-input"
                                    style={
                                      Object {
                                        "display": "inline-block",
                                      }
                                    }
                                  >
                                    <input
                                      aria-activedescendant="react-select-2--value"
                                      aria-expanded="false"
                                      aria-haspopup="false"
                                      aria-owns=""
                                      id="id-undefined"
                                      onBlur={[Function]}
                                      onChange={[Function]}
                                      onFocus={[Function]}
                                      required={false}
                                      role="combobox"
                                      style={
                                        Object {
                                          "boxSizing": "content-box",
                                          "width": "5px",
                                        }
                                      }
                                      value=""
                                    />
                                    <div
                                      style={
                                        Object {
                                          "height": 0,
                                          "left": 0,
                                          "overflow": "scroll",
                                          "position": "absolute",
                                          "top": 0,
                                          "visibility": "hidden",
                                          "whiteSpace": "pre",
                                        }
                                      }
                                    />
                                  </div>
                                </AutosizeInput>
                              </span>
                              <span
                                aria-hidden="true"
                                className="Select-loading-zone"
                              >
                                <span
                                  className="Select-loading"
                                />
                              </span>
                              <span
                                className="Select-arrow-zone"
                                onMouseDown={[Function]}
                              >
                                <span
                                  className="icon-arrow-down"
                                />
                              </span>
                            </div>
                          </div>
                        </Select>
                      </Async>
                    </Component>
                  </MultiSelect>
                </MultiSelectField>
              </SelectOwners>
            </div>
          </Base>
        </Flex>
        <RuleAddButton
          icon="icon-circle-add"
          onClick={[Function]}
          priority="primary"
          size="zero"
        >
          <Button
            className="css-3daogo-RuleAddButton css-1i3lhlw9"
            disabled={false}
            icon="icon-circle-add"
            onClick={[Function]}
            priority="primary"
            size="zero"
          >
            <button
              className="css-3daogo-RuleAddButton css-1i3lhlw9 button button-primary button-zero"
              disabled={false}
              onClick={[Function]}
              role="button"
            >
              <Flex
                align="center"
                className="button-label"
              >
                <Base
                  align="center"
                  className="button-label css-5ipae5"
                >
                  <div
                    className="button-label css-5ipae5"
                    is={null}
                  >
                    <Icon
                      size="zero"
                    >
                      <Base
                        className="css-l3pkcx-Icon e17811v30"
                        size="zero"
                      >
                        <div
                          className="css-l3pkcx-Icon e17811v30"
                          is={null}
                          size="zero"
                        >
                          <StyledInlineSvg
                            size="16px"
                            src="icon-circle-add"
                          >
                            <InlineSvg
                              className="css-1ov3rcq-StyledInlineSvg e17811v31"
                              size="16px"
                              src="icon-circle-add"
                            >
                              <StyledSvg
                                className="css-1ov3rcq-StyledInlineSvg e17811v31"
                                height="16px"
                                viewBox={Object {}}
                                width="16px"
                              >
                                <svg
                                  className="e17811v31 css-1rlza0i-StyledSvg e2idor0"
                                  height="16px"
                                  viewBox={Object {}}
                                  width="16px"
                                >
                                  <use
                                    href="#test"
                                    xlinkHref="#test"
                                  />
                                </svg>
                              </StyledSvg>
                            </InlineSvg>
                          </StyledInlineSvg>
                        </div>
                      </Base>
                    </Icon>
                  </div>
                </Base>
              </Flex>
            </button>
          </Button>
        </RuleAddButton>
      </div>
    </BuilderBar>
  </RuleBuilder>
  <div
    onKeyDown={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <StyledTextArea
      autoCapitalize="off"
      autoComplete="off"
      autoCorrect="off"
      onChange={[Function]}
      placeholder="#example usage
path:src/example/pipeline/* person@sentry.io #infra
url:http://example.com/settings/* #product"
      spellCheck="false"
      value="new"
    >
      <TextareaAutosize
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        className="css-e9agsp-StyledTextArea en3n9di2"
        onChange={[Function]}
        placeholder="#example usage
path:src/example/pipeline/* person@sentry.io #infra
url:http://example.com/settings/* #product"
        rows={1}
        spellCheck="false"
        value="new"
      >
        <textarea
          autoCapitalize="off"
          autoComplete="off"
          autoCorrect="off"
          className="css-e9agsp-StyledTextArea en3n9di2"
          onChange={[Function]}
          placeholder="#example usage
path:src/example/pipeline/* person@sentry.io #infra
url:http://example.com/settings/* #product"
          rows={1}
          spellCheck="false"
          value="new"
        />
      </TextareaAutosize>
    </StyledTextArea>
    <SaveButton>
      <div
        className="css-5zvod-SaveButton en3n9di1"
      >
        <Button
          disabled={false}
          onClick={[Function]}
          priority="primary"
          size="small"
        >
          <button
            className="button button-primary button-sm"
            disabled={false}
            onClick={[Function]}
            role="button"
          >
            <Flex
              align="center"
              className="button-label"
            >
              <Base
                align="center"
                className="button-label css-5ipae5"
              >
                <div
                  className="button-label css-5ipae5"
                  is={null}
                >
                  Save Changes
                </div>
              </Base>
            </Flex>
          </button>
        </Button>
      </div>
    </SaveButton>
  </div>
</OwnerInput>
`;
